sources/common/inc/LogParser.h

```diff
@@ -49,9 +51,15 @@ namespace PARSER
     {
         public:
 
-            /// Returns reference to the signal database.
+            /// Returns the const reference to the signal database.
             const SIGNAL::SignalDb &GetSignalDb() const
             {
+                return const_cast<SIGNAL::SignalDb &>((this)->GetSignalDb());
```

Comment:
Well, we agreed that two get methods are OK so as not to violate existing 'contracts'. E.g. VCDTrace constructor takes the constant reference as it doesn't change anything and only produces the output. On the other hand, the LineCounter needs the non-constant reference as it adds signals to the database.